README


This archive contains the distribution MooseX-Declare,
version 0.42:

  (DEPRECATED) Declarative syntax for Moose

This software is copyright (c) 2008 by Florian Ragwitz.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.036.



Changes
Revision history for MooseX-Declare

0.42      2015-06-04 06:17:43Z
  - eliminate remaining uses of List::MoreUtils
  - increase required version of namespace::autoclean to 0.16 to avoid test
    failures

0.41      2015-05-16 03:35:49Z
  - mark this distribution as deprecated in metadata

0.40      2014-11-12 02:27:58Z
  - mention deprecation status in module ABSTRACT
  - fix warning in tests on 5.8.x

0.39      2014-08-16 19:12:31Z
  - line numbers in shipped code are now almost the same (within 1) as
    the repository source, for easier debugging
  - pod fixes (Frank Wiegand, RT#51722)

0.38      2013-12-14 18:39:30Z
  - update configure_requires checking in Makefile.PL, add CONTRIBUTING file

0.37      2013-09-26 04:08:21Z
  - repository migrated to the github moose organization
  - re-release with fixed compile test

0.36      2013-09-10 02:33:28Z
  - Depend on 'aliased'.
  - The test suite now uses Test::Fatal instead of Test::Exception. (Karen
    Etheridge)
  - removed use of deprecated Class::MOP::load_class, calling enum with a list
  - converted use of Class::MOP::class_of to Moose::Util::find_meta
  - new WARNING section added to pod, regarding future utility of this module

0.35   2011-08-23 11:13:10Z
  - Adapt to changes in MooseX::Role::Parameterized 0.27.

0.34   2010-08-31 23:10:28Z
  * Depend on MooseX::Method::Signatures 0.36 to avoid deprecation warnings on
    new versions of Moose (Jeremy Carman).

0.33   2010-02-04 01:50:14Z
  - Depend on MooseX::Method::Signatures 0.30 to fix various incompatibilities.

0.32   2009-11-02 02:00:42Z
  - Add link to Geany syntax notes to documentation (Michele Beltrame).
  - Convert from MooseX::MethodAttributes to Native::Trait (Justin Hunter).
  - Update dependency on MooseX::Types (Chris Prather).

0.31   2009-09-17 16:10:22Z
  - Bump dependencies for no more warnings on new Moose versions.

0.30   2009-09-05 00:48:05Z
  - Make the attr inheriting TODO test work.
  - Make default values for role parameters work (nperez).
  - Add tests for the above (nperez).

0.29   2009-08-31 17:11:25Z
  - Add failing test for applying modifiers to a method directly composed from
    a role (nperez).
  - Fix the failing test by depending on MXMS 0.24.

0.28   2009-08-27 20:44:34Z
  - Bump prereqs on MooseX::Method::Signatures, Moose, and namespace::clean in
    order to avoid test failure.

0.27   2009-08-18 06:54:22Z
  - Depend on MooseX::Method::Signatures 0.21 to make sure type aliasing works
    and there are no warnings due to the recent MooseX::LazyRequire changes.

0.26   2009-08-16 12:48:12Z
  - Declare dependency on MooseX::AttributeHelpers.

0.25   2009-08-16 03:31:59Z
  - Don't persist the inner keywords over more than one invocation of the declarator.
  - Pass along the current context to import_symbols_from and imported_moose_symbols.
  - Implement parameterizable roles.

0.24   2009-08-15 21:48:09Z
  - Add repository and bugtracker to Makefile.PL (Ash Berlin).
  - Tests and doc fixes on applying multiple roles at once (Chas. J. Owens IV).
  - Produce better diagnostic output on test failures (Dave Rolsky).
  - Make MooseX::Method::Signatures handle the actual parsing for method-like
    keywords (nperez).
  - Clean now warns in autoclean namespaces (Robert 'phaylon' Sedlacek).
  - 'method' keyword now only available by default in namespaces that consumed
    MooseSetup (Robert 'phaylon' Sedlacek).
  - Added 'namespace' keyword and allow namespaces, superclasses and roles to
    be relative (Robert 'phaylon' Sedlacek).
  - Add tests for using anonymous methods as attribute defaults (Devin Austin).
  - Stop depending on B::Hooks::EndOfScope. We don't use it direectly anymore.
  - Add a real 'with' keyword.
  - 'with' in a class- or role-body doesn't defer role application until the
    end of the class- or role-definition anymore. The 'with'-option for the
    class and role keywords still does.
  - Add a tests for role application and method modifier ordering.
  - Make anon classes immutable as well (unless explicitly asked not to).
  - Add a ctx method to add cleanup code that should be injected as early as
    possible.
  - Depend on recent versions of DD and MXMS to make keywords less reserved
    (i.e. hash keys and barewords before fat commas).
  - Make t/with_newlines.t less noisy.
  - Add tests for methods with no signature not caring about their arguments,
    other than $self.

0.23   2009-06-25 03:09:10Z
  - Add test to make sure methods returned from meta are
    MooseX::Method::Signatures::Meta::Method (nperez).
  - Add the actual ::Signature methods to the meta class instead of letting it
    wrap the coderef itself.
  - Make inner scopes inherit the declaration provider instead of hardcoding
    'MooseX::Declare' (Robert 'phaylon' Sedlacek).
  - Fix test failure on new Moose versions, which inline a DESTROY method on
    immutalisation.

0.22   2009-05-06 00:37:37Z
  - Add TODO tests for creating nested anonymous classes and roles
    (nperez).
  - Always end injected code with a semicolon.
  - Only do namespace nesting if the inner namespace starts with "::".
  - Depend on D::D 0.005001 to get a skip_declarator that doesn't fail
    when code is being compiled between calling the linstr callback and
    skip_declarator. This appears to be necessary to work on 5.8.

0.21   2009-05-04 23:55:41Z
  - Add tests for automatic cleaning with the clean/dirty traits
    (Stevan Little).
  - Automatically clean all imports at the end of compile time unless the
    dirty trait is given.
  - Clean now warns in autoclean namespaces (Robert Sedlacek).
  - Document all refactored components (Robert Sedlacek).
  - Improve error reporting (Robert Sedlacek).
  - Added link to emacs syntax highlighter (Nelo Onyiah).
  - Fix a doc typo (reported by Alexander Hartmaier).
  - Make method keyword now only available by default in namespaces that
    consumed MooseSetup (Robert Sedlacek).

Makefile.PL
# This Makefile.PL for MooseX-Declare was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.016
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.34.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "(DEPRECATED) Declarative syntax for Moose",
  "AUTHOR" => "Florian Ragwitz <rafl\@debian.org>",
  "CONFIGURE_REQUIRES" => {
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "MooseX-Declare",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "MooseX::Declare",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Devel::Declare" => "0.005011",
    "Devel::Declare::Context::Simple" => 0,
    "Module::Runtime" => 0,
    "Moose" => "0.90",
    "Moose::Meta::Class" => 0,
    "Moose::Role" => 0,
    "Moose::Util" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "MooseX::Method::Signatures" => "0.36",
    "MooseX::Method::Signatures::Meta::Method" => 0,
    "MooseX::Method::Signatures::Types" => 0,
    "MooseX::Role::Parameterized" => "0.12",
    "MooseX::Types::Moose" => "0.20",
    "Parse::Method::Signatures" => 0,
    "Parse::Method::Signatures::Param::Placeholder" => 0,
    "Sub::Exporter" => 0,
    "Sub::Install" => 0,
    "aliased" => 0,
    "constant" => 0,
    "namespace::autoclean" => "0.16",
    "namespace::clean" => "0.11",
    "overload" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "MooseX::Types::Structured" => 0,
    "Test::Fatal" => 0,
    "Test::Moose" => 0,
    "Test::More" => "0.89",
    "if" => 0,
    "lib" => 0
  },
  "VERSION" => "0.42",
  "test" => {
    "TESTS" => "t/*.t t/recipes/basics/*.t"
  }
);

my %FallbackPrereqs = (
  "Carp" => 0,
  "Devel::Declare" => "0.005011",
  "Devel::Declare::Context::Simple" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Build::Tiny" => "0.039",
  "Module::Runtime" => 0,
  "Moose" => "0.90",
  "Moose::Meta::Class" => 0,
  "Moose::Role" => 0,
  "Moose::Util" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "MooseX::Method::Signatures" => "0.36",
  "MooseX::Method::Signatures::Meta::Method" => 0,
  "MooseX::Method::Signatures::Types" => 0,
  "MooseX::Role::Parameterized" => "0.12",
  "MooseX::Types::Moose" => "0.20",
  "MooseX::Types::Structured" => 0,
  "Parse::Method::Signatures" => 0,
  "Parse::Method::Signatures::Param::Placeholder" => 0,
  "Sub::Exporter" => 0,
  "Sub::Install" => 0,
  "Test::Fatal" => 0,
  "Test::Moose" => 0,
  "Test::More" => "0.89",
  "aliased" => 0,
  "constant" => 0,
  "if" => 0,
  "lib" => 0,
  "namespace::autoclean" => "0.16",
  "namespace::clean" => "0.11",
  "overload" => 0,
  "strict" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
